src/models/BoardGame.py

```diff
@@ -29,3 +29,7 @@ def __post_init__(self):
         else:
```

Comment:
The above sort of feels weird especially using a hard coded string.  But the responsibility for this, because directly related to typing attributes, feels like it should belong in the data class. 

Alternative idea was changing the value of any overallrank to be float like 0 and later on filter out any ranks that are 0.  But the overall rank is not actually 0 a separate caller would have to know to filter out overallranks with 0 otherwise they would be seeing false information.  Which feels even worse than the current implementation.

Reply:
I would gently push back and say that what's going on right now is breaking the type system. The typehint for `overallrank` is `float`, but here you set the value to a string, which goes against the typehint of the attribute.

I know we talked about setting the value of `overallrank` to some special constant if the rank isn't found, and I agree that it pushes the responsibility of filtering to the other class. How about as a compromise you set the `overallrank` to [`math.inf`](https://docs.python.org/3/library/math.html#math.inf) if the rank data does not exist? Since `inf` is greater than every number it should work the same, and then you kill two birds with one stone:
1. The typehint for `overallrank` remains true, regardless of whether or not the data exists in BGG
2. The ranking logic doesn't need to do any filtering and instead just ranks the games it gets on the `overrallrank` attribute

See the following for an example

```python
import math

> a = [7, 4, 10, 1, math.inf]
> a.sort()
> a
>>> [1, 4, 7, 10, inf]
```

Reply:
Hmmmm interesting.  I'm curious how this would look if there's no filtering logic for `inf` values.  Would it look weird in the orderd by overallrank list to have all the `inf` at the top?  I shall give this a go and report back!

Reply:
iiuc it would be the other way around: the board games that have `inf` as their `overallrank` would be at the end of the list.

Reply:
Update : we've finally came up with a way to handle this via a separate module that specifically serializes for json.  This allows us to temporarily set overallranks that are strings to infinity, than later change them to Nones after they've been sorted, so that they are ready to be jsonifyed